qiskit_ibm_runtime/runtime_job.py

```diff
@@ -307,9 +307,17 @@ def _set_status(self, job_response: Dict) -> None:
             IBMError: If an unknown status is returned from the server.
         """
         try:
-            self._status = API_TO_JOB_STATUS[job_response["status"].upper()]
+            if "reason" in job_response["state"]:
+                status = (
+                    job_response["state"]["status"]
+                    + " - "
+                    + job_response["state"]["reason"]
+                )
+            else:
+                status = job_response["state"]["status"]
```

Comment:
This change retrofits the new response format to the existing code in `qiskit-ibm-runtime`.

I would prefer if we could keep our implementation aligned with the API changes, i.e. use separate properties to store the returned `reason` and `status` values instead of string concatenation.

As a result we would need to infer the job status in a slightly different way. Instead of accessing the [key-value maps](https://github.com/Qiskit/qiskit-ibm-runtime/blob/9f8b89e5de71a93b1ebc49d94bac6fdcea6af58f/qiskit_ibm_runtime/constants.py#L20-L33) there could be a dedicated helper functions to compute the job status or job error message given the returned state (reason/status) from the API.

Reply:
Competing opinions, @kt474 or @rathishcholarajan?

Reply:
Yes I'd prefer storing `status` and `reason` as separate properties as well. We might expand to giving reasons for other sorts of failures.